electrum/gui/qt/paytoedit.py

```diff
@@ -135,6 +135,8 @@ def parse_amount(self, x):
             return '!'
         p = pow(10, self.amount_edit.decimal_point())
         try:
+            if check_max_spend(x): #check for max spend
+                return str(parse_max_spend(x)) + '!'
```

Comment:
If using the changed `parse_max_spend` I suggest in another comment,
you could remove these two lines,
and simply change the `if x == '!': return '!'` test above
to `if parse_max_spend(x): return x`